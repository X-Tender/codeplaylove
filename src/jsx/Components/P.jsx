import React from 'react';
import PropTypes from 'prop-types';

function P({ children, className }) {
	const classes = `paragraph ${className}`;
	return <p className={classes}>{children}</p>;
}

P.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
};

P.defaultProps = {
	children: null,
	className: '',
};

export default P;
