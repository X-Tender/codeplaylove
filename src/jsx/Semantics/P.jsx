import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
};

const defaultProps = {
	children: null,
	className: '',
};

const P = ({ children, className }) => {
	const classes = `paragraph ${className}`;
	return <p className={classes}>{children}</p>;
};

P.propTypes = propTypes;
P.defaultProps = defaultProps;

export default P;
