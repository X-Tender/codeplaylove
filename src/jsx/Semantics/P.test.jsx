import React from 'react';
import { shallow } from 'enzyme';
import P from './P';

describe('P', () => {
	const className = 'paragraph';
	const children = <span>Don&#39;t you, forget about me</span>;

	describe('with all props', () => {
		const comp = shallow(<P className={className}>{children}</P>);

		it('renders correctly', () => {
			expect(comp).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			expect(comp.find('p').hasClass(className)).toBe(true);
			expect(comp.contains(children)).toBe(true);
		});
	});

	describe('with no props', () => {
		const comp = shallow(<P />);

		it('renders correctly', () => {
			expect(comp).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			expect(comp.children()).toHaveLength(0);
		});
	});
});
