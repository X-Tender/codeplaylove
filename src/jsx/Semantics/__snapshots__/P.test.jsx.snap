// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P with all props renders correctly 1`] = `
<p
  className="paragraph paragraph"
>
  <span>
    Don't you, forget about me
  </span>
</p>
`;

exports[`P with no props renders correctly 1`] = `
<p
  className="paragraph "
/>
`;
